src/sql/parts/objectExplorer/common/objectExplorerService.ts

```diff
@@ -320,41 +356,95 @@ export class ObjectExplorerService implements IObjectExplorerService {
 				let newRequest = false;
 				if (!self._sessions[session.sessionId].nodes[nodePath]) {
 					self._sessions[session.sessionId].nodes[nodePath] = {
-						expandEmitter: new Emitter<sqlops.ObjectExplorerExpandInfo>()
+						expandEmitter: new Emitter<NodeExpandInfoWithProviderId>()
 					};
 					newRequest = true;
 				}
-				self._sessions[session.sessionId].nodes[nodePath].expandEmitter.event(((expandResult) => {
-					if (expandResult && !expandResult.errorMessage) {
-						resolve(expandResult);
-					}
-					else {
-						reject(expandResult ? expandResult.errorMessage : undefined);
+				let provider = this._providers[providerId];
+				if (provider) {
+					let resultMap: Map<string, sqlops.ObjectExplorerExpandInfo> = new Map<string, sqlops.ObjectExplorerExpandInfo>();
+					let allProviders: sqlops.ObjectExplorerProviderBase[] = [provider];
+
+					let nodeProviders = this._nodeProviders[providerId];
+					if (nodeProviders) {
+						nodeProviders = nodeProviders.sort((a, b) => a.group.toLowerCase().localeCompare(b.group.toLowerCase()));
+						allProviders.push(...nodeProviders);
 					}
+
+					self._sessions[session.sessionId].nodes[nodePath].expandEmitter.event((expandResult) => {
+						if (expandResult) {
```

Comment:
You've missed out on the ` && !expandResult.errorMessage` clause. This will lose failure results coming back. We need to fix this and handle the case where 1 provider has an error, but the other doesn't. Or is this handled later in the process?
- At a minimum add a comment explaining why we're not checking for error message here.